src/WeatherSensor.cpp

```diff
@@ -833,21 +828,21 @@ DecodeStatus WeatherSensor::decodeBresser7In1Payload(uint8_t *msg, uint8_t msgSi
 //  if (status != DECODE_OK)
```

Comment:
`status` is updated by `findSlot()`; it returns `DECODE_OK` if decoding was successful (up to this point) and the sensor data could be stored; i.e. it passed the ID filtering and a free slot was found.

Only in this case `sensor[slot]` may be updated.